src/main/java/ch/njol/skript/expressions/ExprBannerPatterns.java

```diff
@@ -0,0 +1,291 @@
+		return switch (mode) {
+			case SET -> banner -> {
+				banner.setPatterns(patterns);
+			};
+			case DELETE -> banner -> {
+				banner.setPatterns(new ArrayList<>());
+			};
+			case ADD -> banner -> {
+				patterns.forEach(banner::addPattern);
+			};
+			case REMOVE -> banner -> {
+				List<Pattern> current = banner.getPatterns();
+				current.removeAll(patterns);
+				banner.setPatterns(current);
+			};
+			default -> banner -> {};
+		};
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE -> {
+				if (patternNumber != null)
+					yield CollectionUtils.array(Pattern.class);
+				yield CollectionUtils.array(Pattern[].class);
+			}
+			case REMOVE, ADD -> {
+				if (patternNumber != null)
+					yield null;
+				yield CollectionUtils.array(Pattern[].class);
+			}
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		Pattern[] patterns = (Pattern[]) delta;
+		int placement = 0;
+		if (patternNumber != null)
+			placement = patternNumber.getSingle(event);
+		List<Pattern> patternList = patterns != null ? Arrays.stream(patterns).toList() : new ArrayList<>();
+
+		Consumer<BannerMeta> metaChanger = null;
+		Consumer<Banner> blockChanger = null;
+
+		if (placement >= 1) {
+			Pattern pattern = patternList.size() == 1 ? patternList.get(0) : null;
+			metaChanger = getPlacementMetaChanger(mode, placement, pattern);
+			blockChanger = getPlacementBlockChanger(mode, placement, pattern);
+		} else {
+			metaChanger = getAllMetaChanger(mode, patternList);
+			blockChanger = getAllBlockChanger(mode, patternList);
+		}
+
+		for (Object object : objects.getArray(event)) {
+			if (object instanceof Block block && block.getState() instanceof Banner banner) {
+				blockChanger.accept(banner);
+				banner.update(true, false);
+			} else {
+				ItemStack itemStack = ItemUtils.asItemStack(object);
+				if (itemStack == null || !(itemStack.getItemMeta() instanceof BannerMeta bannerMeta))
+					continue;
+				metaChanger.accept(bannerMeta);
+				itemStack.setItemMeta(bannerMeta);
+				if (object instanceof Slot slot) {
+					slot.setItem(itemStack);
+				} else if (object instanceof ItemType itemType) {
+					itemType.setItemMeta(bannerMeta);
+				} else if (object instanceof ItemStack itemStack1) {
+					itemStack1.setItemMeta(bannerMeta);
+				}
+			}
+		}
+
+	}
+
+	@Override
+	public boolean isSingle() {
+		return patternNumber != null && getExpr().isSingle();
+	}
+
+	@Override
+	public Class<Pattern> getReturnType() {
+		return Pattern.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		SyntaxStringBuilder builder = new SyntaxStringBuilder(event, debug);
+		if (patternNumber != null) {
+			builder.append(patternNumber, "banner pattern");
+		} else {
+			builder.append("banner patterns");
+		}
+		builder.append(objects);
```

Comment:
I don't think this toString makes very much sense. Couldnt a possible return be `1 banner pattern player's tool`
I think it should be something more like `banner pattern (number) of (objects)` or `banner patterns of (objects)`